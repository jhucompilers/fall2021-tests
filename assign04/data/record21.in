1
5 8 2
2 4 1
99 23 6
