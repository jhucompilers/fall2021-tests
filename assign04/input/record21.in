PROGRAM record21;
  -- record type containing 2D array
  TYPE Board = RECORD
    turn : INTEGER;
    grid : ARRAY 3 OF ARRAY 3 OF INTEGER;
  END;
  VAR board: Board;
  VAR i, j: INTEGER;

BEGIN
  READ board.turn;

  i := 0;
  WHILE i < 3 DO
    j := 0;
    WHILE j < 3 DO
      READ board.grid[i, j];
      j := j + 1;
    END;
    i := i + 1;
  END;

  WRITE (board.turn + 1) MOD 2;

  i := 0;
  WHILE i < 3 DO
    j := 0;
    WHILE j < 3 DO
      WRITE board.grid[i, j] * 2;
      j := j + 1;
    END;
    i := i + 1;
  END;
END.
