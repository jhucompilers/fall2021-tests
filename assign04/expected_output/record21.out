0
10
16
4
4
8
2
198
46
12
